scripts/Packages/Package_Gullwing__QFP_SOIC_SO/Readme.md

```diff
@@ -0,0 +1,121 @@
+
+``` yaml
+internal_package_name:
+  # parameter list (See below)
+```
+
+### Documentation and naming parameters
+- Size source to be added to footprint documentation field (`size_source`) {url}
+- footprint name control
+  - Manufacturer if specific to one manufacturer (`manufacturer`) {string}
+  - Part number if specific to one part (`part_number`) {string}
+  - Custom naming (`custom_name_format`) {format string}
+
+### Package dimensions
+![dimension example](./documentation/dimension_system.svg)
+- Body size (`body_size_x`, `body_size_y`, `overall_height`) {dimension}
+- Lead dimensions:
+  - Over all size representing lead tip to lead tip dimension (`overall_size_x`, `overall_size_y`) {dimension}
+  - Lead with (`lead_width`) {dimension}
+  - Lead length representing tip till bend (`lead_len`)
+- lead pitch, currently equal for all sides (`pitch`) {float}
+- pin count, 0 means no pins in that direction (`num_pins_x`, `num_pins_y`) {int}
+
+
+### Exposed pad handling:
+![exposed pad example](../documentation/ep_handling.svg)
+- Size of package exposed pad or slug (`EP_size_x`, `EP_size_y`) {dimension}
+- Optional the size of the footprint pad (`EP_size_x_overwrite`, `EP_size_y_overwrite`) {float}
+   - Pad size is equal to nominal package pad size if not given.
+- Optional the size of the mask cutout (`EP_mask_x`, `EP_mask_y`) {float}
+   - Use to create solder mask defined pads (in combination with `EP_size_x_overwrite`)
+- Paste is typically split into a regular grid with (`EP_num_paste_pads`) {[int (x), int (y)]}
+  - The paste coverage multiplicator determines how much of the exposed copper area is covered by paste. (`EP_paste_coverage`) {float (0..1), default=0.65}
+
+### Thermal vias
+A package with exposed pad can generate a version with thermal vias. This will be generated in addition to the normal version.
+
+![exposed pad example](../documentation/thermal_vias.svg)
+
+``` yaml
+  thermal_vias:
+    # thermal via version parameters
+```
+- Number of vias generated in the regular grid (`count`) {[int (x), int (y)]}
+- Final hole size (`drill`) {float}
+- Optional grid (`grid`) {[float (x), float (y)]}
+  - Auto generated if not given (outermost pad will touch pad edge.)
+- Optional paste coverage overwrite (`EP_paste_coverage`) {float (0..1)}
+  - Increase compared to non thermal via version to combat solder loss through wicking.
+- Should paste avoid vias? (`paste_avoid_via`) {bool}
+  - Can lead to math exceptions (reduce paste coverage)
+  - Clearance between via hole and paste controlled with (`paste_via_clearance`) {float}
+- Number paste pads
+  - given as count value (`EP_num_paste_pads`) {[int (x), int (y)]}
+  - More specific as between vias and outside ring (`paste_between_vias`, `paste_rings_outside`)
+
+
+## Dimension parameter format
+
+### String based
```

Comment:
I think a hyphen is missing and should be `String-based`. Same as for the dict format below.

Reply:
done